### PR TITLE
feat: ssr-using-java

### PR DESCRIPTION
**Pull request by mswertz**
To test:

Server side rendering using GraalVM:

pattern http://server/:schema/view/:table/:filter/:view
e.g.
https://preview-pr-emx2-850.dev.molgenis.org/pet%20store/view/Pet/{"name":"spike"}/json
https://preview-pr-emx2-850.dev.molgenis.org/pet%20store/view/Pet/{"name":"pooky"}/details

For development you can also run using 'yarn serve' in package 'ssr'
e.g.
http://localhost:9091/#/Pet/{%22name%22:%22spike%22}/json

Obviously, the 'filter' part of the URL is not very elegant. Proposed solutions:
(1) we will give a automatic ID to all records in emx2 so we can simply retrieve by id
(2) we will implement custom routes. In current setup those then need also implementation in Java which is not pretty

Other remaining issues:
* adding a html around the result, plus a MolgenisWrapper
* decide if and how we want to make 'styleguide' compatible with ssr

### CODE REVIEW

**Comment by connoratrug** [comment]
[Have server render (initial) page requests aka SSR](https://trello.com/c/kiEVYZ8T/362-have-server-render-initial-page-requests-aka-ssr)

**Comment by mswertz** [comment]
for some odd reason many more files are in this pull request then needed. Will investigate what happened

File: backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/SsrService.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+public class SsrService {
+
+  static Logger logger = LoggerFactory.getLogger(SsrService.class);
+
+  public static String SSR_BASE_PATH = "/public_html/ssr/";
+
+  public static void create() {
+    get("/:schema/view/:table/:filter", SsrService::getRecordForDefaultRoute);
+    get("/:schema/view/:table/:filter/:route", SsrService::getRecordForRoute);
```
Comment (connoratrug) [inline backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/SsrService.java:38]: having global path fiters spread across the code base makes it hard is evaluate if path are being missed due to other paths filters being greedy, suggest to the these in the main server class of in a separate route class

File: settings.gradle [hunk 2]
```diff
@@ -11,6 +11,7 @@ include(':apps:schema')
 include(':apps:updownload')
 include(':apps:tables')
 include(':apps:pages')
+include(':apps:ssr')
```
Comment (connoratrug) [inline settings.gradle:14]: suggest to change the name to 'graal-ssr' or 'java-ssr'

**Review by connoratrug — APPROVED** [review]
not sure if this approach scales well ( adding more js views and having multiple devs work on frontend), but the code works and does not break other features, suggest to merge as poc after updating module name

**Comment by connoratrug** [comment]
note up to date with main branch , closing , please reopen if needed

**Pull request closed without merging**